In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/facebook-ad-campaign/data.csv


In [2]:
dataset = pd.read_csv("/kaggle/input/facebook-ad-campaign/data.csv")
len(dataset)

1143

# **Computing Number of impressions in Age range 30-34**

In [4]:
ageRestrictedDataset =dataset[dataset.age == "30-34"] 
totalImpressions = ageRestrictedDataset.impressions.sum()
print(f"Total Impressions in age between 30-34 are : {totalImpressions}.")

Total Impressions in age between 30-34 are : 35678593.0.


**Total Impressions in age between 30-34 are : 35678593.0**

### **converting reporting_start from string to Date and time.** 

In [6]:
from dateutil import parser
for x in range(0,len(dataset)):
    dataset.reporting_start[x] = parser.parse(dataset.reporting_start[x])
dataset

/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


ad_id      reporting_start reporting_end campaign_id fb_campaign_id  \
0      708746  2017-08-17 00:00:00    17/08/2017         916         103916   
1      708749  2017-08-17 00:00:00    17/08/2017         916         103917   
2      708771  2017-08-17 00:00:00    17/08/2017         916         103920   
3      708815  2017-08-30 00:00:00    30/08/2017         916         103928   
4      708818  2017-08-17 00:00:00    17/08/2017         916         103928   
...       ...                  ...           ...         ...            ...   
1138  1314410  2017-08-19 00:00:00    19/08/2017       45-49              F   
1139  1314411  2017-08-19 00:00:00    19/08/2017       45-49              F   
1140  1314412  2017-08-19 00:00:00    19/08/2017       45-49              F   
1141  1314414  2017-08-17 00:00:00    17/08/2017       45-49              F   
1142  1314415  2017-08-17 00:00:00    17/08/2017       45-49              F   

        age gender  interest1  interest2  interest3   impressions  clicks  \
0     30-34      M         15         17         17   7350.000000       1   
1     30-34      M         16         19         21  17861.000000       2   
2     30-34      M         20         25         22    693.000000       0   
3     30-34      M         28         32         32   4259.000000       1   
4     30-34      M         28         33         32   4133.000000       1   
...     ...    ...        ...        ...        ...           ...     ...   
1138    109    111        114    1129773        252    358.189997      13   
1139    110    111        116     637549        120    173.880003       3   
1140    111    113        117     151531         28     40.289999       2   
1141    113    114        117     790253        135    198.710001       8   
1142    114    116        118     513161        114    165.609999       5   

      spent  total_conversion  approved_conversion  
0      1.43               2.0                  1.0  
1      1.82               2.0                  0.0  
2      0.00               1.0                  0.0  
3      1.25               1.0                  0.0  
4      1.29               1.0                  1.0  
...     ...               ...                  ...  
1138   2.00               NaN                  NaN  
1139   0.00               NaN                  NaN  
1140   0.00               NaN                  NaN  
1141   2.00               NaN                  NaN  
1142   2.00               NaN                  NaN  

[1143 rows x 15 columns]

### **Computing Number of clicks from date range 19/08/2017 - 22/08/2017**

In [8]:
from dateutil import parser
startDate = parser.parse('19/08/2017')
endDate = parser.parse('22/08/2017')
dateRestrictedDataset = dataset[dataset.reporting_start<= endDate]
len(dateRestrictedDataset)
# dateRestrictedDataset
dateRestrictedDataset = dateRestrictedDataset[dateRestrictedDataset.reporting_start>=startDate]
print(f"total clicks in date range 19/08/2017 - 22/08/2017 are : {dateRestrictedDataset.clicks.sum()}")

total clicks in date range 19/08/2017 - 22/08/2017 are : 3763


**total clicks in date range 19/08/2017 - 22/08/2017 are : 3763**

### **All ad_ids for every campaign id**

In [12]:
for id in dataset.campaign_id.unique() :
    ad_id = dataset[dataset.campaign_id==id].ad_id
    print(f"for campaign id {id} ad ids are :{ad_id}")

for campaign id 916 ad ids are :0     708746
1     708749
2     708771
3     708815
4     708818
5     708820
6     708889
7     708895
8     708953
9     708958
10    708979
11    709023
12    709038
13    709040
14    709059
15    709105
16    709115
17    709124
18    709179
19    709183
20    709320
21    709323
22    709326
23    709327
24    709328
25    709455
26    709544
27    709614
28    709756
29    709761
30    709899
31    709901
32    710045
33    710088
34    710360
35    710477
36    710480
37    710571
38    710617
39    710623
40    710628
41    710682
42    710763
43    710836
44    710867
45    710880
46    710961
47    710968
48    711217
49    711623
50    711764
51    711785
52    711877
53    712052
Name: ad_id, dtype: int64
for campaign id 936 ad ids are :54     734209
55     734210
56     734215
57     734243
58     734266
        ...  
513    951941
514    952001
515    952031
516    952080
517    952100
Name: ad_id, Length: 464, dtype: int64
for campaign id